bin/build_cmd.ml

```diff
@@ -30,6 +30,38 @@ let with_metrics ~common f =
     Fiber.return ())
 ;;
 
+let raise_on_lock_dir_out_of_sync () =
+  Memo.of_thunk (fun () ->
```

Comment:
This one needs to be `Memo.lazy_` to avoid recomputation I believe

Reply:
I don't have enough knowledge on `Memo`: what is the difference from `Memo.lazy_` and `Memo.thunk` (apart the fact that the first one wrap the result in a Lazy structure)? From my previous understanding, the role of `Memo` was to ensure there isn't some "action duplication". What do I miss here? :confused:

Reply:
The difference is as how bind behaves on them. When you bind on `Memo.of_thunk f`, `f` will always be recomputed on demand. While if you bind on `Memo.Lazy.force (Memo.lazy_ f))`, you will only recompute `f` when necessary.

Reply:
Thanks for the explanation :pray: